lib/temporal/metadata/activity.rb

```diff
@@ -23,6 +23,19 @@ def initialize(namespace:, id:, name:, task_token:, attempt:, workflow_run_id:,
       def activity?
         true
       end
+
+      def to_h
+        {
+          'namespace' => namespace,
+          'workflow_id' => workflow_id,
+          'workflow_name' => workflow_name,
+          'workflow_run_id' => workflow_run_id,
+          'activity_id' => id,
+          'activity_name' => name,
+          'attempt' => attempt,
+          'heartbeat_details' => heartbeat_details
```

Comment:
If we're leaving `heartbeat_details`, should we be concerned with decoding the information in them? You'd expect plain values in this hash, but `heartbeat_details` does sound like it is it's own world